src/library_gl.js

```diff
@@ -1023,6 +1023,13 @@ var LibraryGL = {
 
   glCompressedTexImage2D__sig: 'viiiiiiii',
   glCompressedTexImage2D: function(target, level, internalFormat, width, height, border, imageSize, data) {
+#if USE_WEBGL2
+    if (GL.currentContext.version >= 2) { // WebGL 2 provides new garbage-free entry points to call to WebGL. Use those always when possible.
```

Comment:
We could maybe do even better, avoiding this `if` on every call by replacing the function with a new one. However, that might be overkill, and is messy. Maybe at some point we could profile it though.